docs/streams/upgrade-guide.html

```diff
@@ -38,6 +44,16 @@
         See <a href="#streams_api_changes_0101">below</a> a complete list of 0.10.1 API changes that allow you to advance your application and/or simplify your code base, including the usage of new features.
     </p>
 
+    <h3><a id="streams_api_changes_100" href="#streams_api_changes_100">Streams API changes in 1.0.0</a></h3>
+    <!-- TODO: KIP-120, 138, 160, 161, 167, 173 -->
```

Comment:
This PR is just for setting up the section, and only add the metrics changes.

The plan is to let each KIP contributor to add to this section in separate following pRs.